### ISSUE TITLE
Resizing a panel sometimes gets stuck dragging forever

### ISSUE THREAD

**Issue by mpearson** [mathuo/dockview#249]
**Describe the bug**
When you start resizing a panel, the mouse sometimes changes to drag and drop cursor instead of staying on the resize cursor. If this happens, releasing the mouse no longer stops the drag operation, and the divider will continue following the mouse until you click again. Seems like a native drag and drop event is not being cancelled or something. This happens maybe 1 in 5 attempts to resize, although sometimes it doesn't happen at all and other times it happens on almost every attempt.

I mostly see this in Chrome although it does happen less frequently in Firefox. Seen on both Windows and Ubuntu as well.

**To Reproduce**
1. Mouse down on any resizable panel border
2. Move mouse
3. Get unlucky

**Expected behavior**
Resize operation should stop when the mouse is released. Resize mouse cursor should be displayed the whole time.

**Environment:**
 - Chrome 112.0.5615.138, Firefox 112.0.2
 - Windows 10, Ubuntu 18.04

**Screen Capture**
This screenshot shows the drag-and-drop cursor that appears when this bug happens.
The video shows the drag operation getting stuck. Enable audio to hear when the clicks start and stop:
![Image1](https://user-images.githubusercontent.com/926478/236610335-9cc60ad8-66b8-4f0c-ac99-02ac7769d6cd.png)

[Recording showing the broken drag occurring twice.](https://user-images.githubusercontent.com/926478/236609602-64bd8656-c700-4872-8193-6c7a03700250.mp4)

**Comment by mpearson** [comment]
Good news! I think I found a workaround. I noticed when getting stuck in drag mode, I could see the title of one of the panels very faintly, moving around with the mouse:

![Image2](https://user-images.githubusercontent.com/926478/236612517-b3f9b217-8e10-460f-b604-13604bb03b3f.png)

This made me think the entire panel element was being dragged. I added the following CSS rule and have been unable to recreate the bug since. Note that both selectors seem to be required, otherwise it occasionally reproduces, so I guess both of these elements are sometimes getting selected.

```
.split-view-container .view-container .view, .groupview {
  user-select: none;
}
```

**Comment by mpearson** [comment]
Realized that the previous CSS rule was breaking *all* text selection inside the panel, which is obviously not ideal. Worse, re-enabling `user-select: auto` on any child element seemed to break the whole thing and the bug came back. Finally I figured out that it just needs to be applied to the drag handles themselves:

```
.split-view-container .sash-container .sash {
  user-select: none;
  -webkit-user-select: none; /* Safari */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* IE 10 and IE 11 */
}
```

This appears to fix the bug as well as preserve normal selection of panel content.

**Comment by mathuo** [comment]
Thanks for the detailed description and the attempt to fix it too!

I just checked this out by adding a global drag listener to check what was actually being dragged here, something like:

```js
document.addEventListener('dragStart', (e) => {
    console.log('drag', e)
});
```
and yes it seems that the *sash* element is unintentionally entering a drag state occasionally as below. I think your analysis and fix is correct, great spot.

![screenshot](https://user-images.githubusercontent.com/6710312/236630252-1ee98323-deb5-425b-8495-3c37b08f5b7a.png)

I will double-check a few things and aim to merge your PR and publish a fix version for this in the next day or two.

Thanks for finding the bug and the fix :)

**Comment by mathuo** [comment]
This is now merged and published as v1.7.2

**Comment by mpearson** [comment]
Awesome, thanks for the quick turnaround! Was really excited about using this library caus it's great. Glad the solution didn't end up being too complicated.
